api/app/registrationResponse/api.py

```diff
@@ -0,0 +1,239 @@
+import flask_restful as restful
+from flask import g, request
+from sqlalchemy.exc import SQLAlchemyError
+from app.utils.auth import verify_token
+from app.registrationResponse.mixins import RegistrationResponseMixin
+from app.registration.models import Offer, Registration, RegistrationAnswer, RegistrationForm, RegistrationQuestion
+from app.users.models import AppUser
+from app.events.models import Event
+from app.utils.auth import auth_required
+from app.utils import errors, emailer, strings
+from app import LOGGER
+from app.users.repository import UserRepository as user_repository
+
+
+from app import db
+
+
+def _get_answer_value(answer, question):
+    if question.type == 'multi-choice' and question.options is not None:
+        value = [o for o in question.options if o['value'] == answer.value]
+        if not value:
+            return answer.value
+        return value[0]['label']
+
+    if question.type == 'file' and answer.value:
+        return 'Uploaded File'
+
+    return answer.value
+
+
+class RegistrationApi(RegistrationResponseMixin, restful.Resource):
+    answer_fields = {
+        'id': fields.Integer,
+        'registration_id': fields.Integer,
+        'registration_question_id': fields.Integer,
+        'value': fields.String
+    }
+
+    registration_fields = {
+        'id': fields.Integer,
+        'offer_id': fields.Integer,
+        'registration_form_id': fields.Integer,
+        'confirmed': fields.Boolean,
+        'created_at': fields.DateTime,
+        'confirmation_email_sent_at': fields.DateTime
+
+    }
+
+    response_fields = {
+
+        'registration_id': fields.Integer,
+        'offer_id': fields.Integer,
+        'registration_form_id': fields.Integer,
+        'answers': fields.List(fields.Nested(answer_fields))
+    }
+
+    @marshal_with(response_fields)
+    @auth_required
+    def get(self):
+
+        try:
+            user_id = verify_token(request.headers.get('Authorization'))['id']
+
+            db_offer = db.session.query(Offer).filter(Offer.user_id == user_id).first()
+
+            if db_offer is None:
+                return errors.OFFER_NOT_FOUND
+            registration = db.session.query(Registration).filter(Registration.offer_id == db_offer.id).first()
+
+            if registration is None:
+                return 'no Registration', 404
+            registration_form = db.session.query(RegistrationForm).filter(RegistrationForm.id == registration.
+                                                                          registration_form_id).first()
+
+            if registration_form is None:
+                return errors.REGISTRATION_FORM_NOT_FOUND
+            db_answers = db.session.query(RegistrationAnswer).filter(RegistrationAnswer.registration_id ==
+                                                                     registration.id).all()
+
+            response = {
+                'registration_id': registration.id,
```

Comment:
Please also add "Registration.confirmed" and the created_at datetime to the response